runtime/storage.go

```diff
@@ -283,117 +177,143 @@ func (s *Storage) recordContractUpdate(
 			panic(err)
 		}
 
-		s.contractUpdates[storageKey] = storable
+		s.contractUpdates[key] = storable
```

Comment:
Not sure, but is there a leak here?

Contract storable is created and stored in `contractUpdates` map in `recordContractUpdate()`.  In `writeContractUpdate()` contract storable is converted to Value object and is inserted to underlying ordered map.  Ordered map creates another storable for this contract during insertion.

Reply:
Yes, good catch! I guess there is no need to keep contract updates as storables, they can just be kept as values in `contractUpdates`, and then `writeContractUpdate` implicitly gets the storable during the insertion into the storage map? I've refactored this in d3a4ef2efc4e82c2825a9bb39960a48ce1277e48, what do you think?

Reply:
LGTM!  :+1: